src/actions/airtable/airtable.ts

```diff
@@ -66,37 +82,42 @@ export class AirtableAction extends Hub.OAuthAction {
         })
       }))
     } catch (e: any) {
-      response = { success: false, message: e.message }
+      response.success = false
+      response.message = e.message
     }
+    response.state = state
     return new Hub.ActionResponse(response)
   }
 
-  async checkBaseList(request: Hub.ActionRequest) {
-    if (request.params.state_json) {
-      const stateJson = JSON.parse(request.params.state_json)
-      const response = await this.refreshTokens(stateJson.tokens.refresh_token)
-      return gaxios.request({
-        method: "GET",
-        url: "https://api.airtable.com/v0/meta/bases",
-        headers: {
-          Authorization: `Bearer ${(response.data as any).access_token}`,
-        },
-      }).catch((_err) => {
-        throw "Error listing bases, oauth credentials most likely expired."
-      })
-    } else {
-      return null
-    }
+  async checkBaseList(token: String) {
+    return gaxios.request({
+      method: "GET",
+      url: "https://api.airtable.com/v0/meta/bases",
+      headers: {
+        Authorization: `Bearer ${token}`,
+      },
+    }).catch((_err) => {
+      throw "Error listing bases, oauth credentials most likely expired."
+    })
   }
 
   async form(request: Hub.ActionRequest) {
     const form = new Hub.ActionForm()
     try {
-      const response = await this.checkBaseList(request)
-      if (response === null) {
-        // @ts-ignore
-        throw "Error with checking baselist"
+      let accessToken
+      if (request.params.state_json) {
```

Comment:
nit: this part of the method could be taken out of both form and execute to become `retrieveToken` or something, because this is duplicated between the two at the moment.